### ISSUE TITLE
2015.2.0rc1: zfs errors in log after update

### ISSUE THREAD

**Issue by dennisjac** [saltstack/salt#20635]
After updating to 2015.2.0rc1 I see strange error messages in the log:
```
[ERROR   ] Command 'modinfo zfs' failed with return code: 1
[ERROR   ] output: ERROR: modinfo: could not find module zfs
[ERROR   ] Command 'modinfo zfs' failed with return code: 1
[ERROR   ] output: ERROR: modinfo: could not find module zfs
```

We are not using zfs anywhere so I'm not sure why this gets reported by saltstack.

**Comment by jfindlay** [comment]
@dennisjac, thanks for the report.

**Comment by jfindlay** [comment]
@dennisjac, what system are you seeing this on?

**Comment by rquinnlv** [comment]
Found same issue on Ubuntu 14.04

**Comment by jfindlay** [comment]
@dennisjac, @rquinnlv, do you have anything special in your configs or setups otherwise?  I'm trying to narrow down this issue but cannot reproduce it or reason why the error would repeat at all.

**Comment by jfindlay** [comment]
@jacksontj, @cachedout could this be related to recent loader changes?

**Comment by jacksontj** [comment]
This is actually unrelated to the loader change, this is due to a change in the zfs module (da6395f) which is now shelling out to find zfs. I pushed a fix in 13a8d15

**Comment by jfindlay** [comment]
@jacksontj, nice thanks.

**Comment by jfindlay** [comment]
@dennisjac, @rquinnlv, do you still see this error after upgrading to the latest 2015.2?

**Comment by thatch45** [comment]
I am still seeing these in develop

**Comment by jacksontj** [comment]
My change was merged into 2015.2 was it merged up to develop yet?

**Comment by thatch45** [comment]
Based on what I am seeing that is the case, so we just need to merge forward :)

**Comment by miihael** [comment]
This still throws an error if "modinfo" command is not present on the system:

    Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1176, in process_virtual
        virtual = mod.__virtual__()
      File "/usr/lib/python2.7/dist-packages/salt/modules/zfs.py", line 86, in __virtual__
        if cmd and salt_cmd.retcode(cmd, output_loglevel='quiet') == 0:
      File "/usr/lib/python2.7/dist-packages/salt/modules/cmdmod.py", line 1099, in retcode
        use_vt=use_vt)
      File "/usr/lib/python2.7/dist-packages/salt/modules/cmdmod.py", line 391, in _run
        .format(cmd, kwargs, exc)
    CommandExecutionError: Unable to run command ['modinfo', 'zfs'] with the context {'with_communicate': True, 'shell': False, 'env': {'LC_ALL': 'C', 'SHELL': '/bin/sh', 'CONSOLE': '/dev/console', 'SHLVL': '4', 'runlevel': '3', 'LINES': '24', 'PREVLEVEL': 'N', 'PATH': '/sbin:/usr/sbin:/bin:/usr/bin', 'COLUMNS': '80', 'RUNLEVEL': '3', 'container': 'lxc', 'INIT_VERSION': 'sysvinit-2.88', 'HOME': '/root', '_': '/usr/bin/python2.7', 'previous': 'N'}, 'stdout': -1, 'close_fds': True, 'stdin': None, 'stderr': -2, 'cwd': '/root'}, reason: [Errno 2] No such file or directory

salt version is 2015.5.0

**Comment by jfindlay** [comment]
@miihael, the change I've made should fix the error you found.  This fix will come out in 2015.5.2 in about two weeks.
